File: 프로그래머스/0/120875. 평행/평행.py
def solution(dots):
    def is_parallel(a, b, c, d):
        return (b[1] - a[1]) * (d[0] - c[0]) == (d[1] - c[1]) * (b[0] - a[0])

    pairs = [
        (0, 1, 2, 3),
        (0, 2, 1, 3),
        (0, 3, 1, 2)
    ]

    for a, b, c, d in pairs:
        if is_parallel(dots[a], dots[b], dots[c], dots[d]):
            return 1
    return 0

File: 프로그래머스/0/120875. 평행/평행.js
function solution(dots) {
  const [a, b, c, d] = dots;

  const slope1 = (b[1] - a[1]) / (b[0] - a[0]);
  const slope2 = (d[1] - c[1]) / (d[0] - c[0]);
  if (slope1 === slope2) return 1;

  const slope3 = (c[1] - a[1]) / (c[0] - a[0]);
  const slope4 = (d[1] - b[1]) / (d[0] - b[0]);
  if (slope3 === slope4) return 1;

  const slope5 = (d[1] - a[1]) / (d[0] - a[0]);
  const slope6 = (c[1] - b[1]) / (c[0] - b[0]);
  if (slope5 === slope6) return 1;

  return 0;
}
